### PR TITLE
Update Copyright notice to remove year

### PR DESCRIPTION
**Pull request by kwiknick**
This change is to fix Issue #3148 - Update copyright notices. 
Changed the ManagedEntranceStrings resource file to take in a parameter.  Then edited the ManagedEntrance.cs to grab the year using GetDate.Now and placing it in the banner variable for display when powershell is opened.

### CODE REVIEW

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs [hunk 1]
```diff
@@ -106,12 +106,13 @@ public int Start(string consoleFilePath, string[] args)
                     RunspaceConfigForSingleShell.Create(consoleFilePath, out warning);
             }
             int exitCode = 0;
+            string systemYear = DateTime.Now.Year.ToString();
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs]: Do we really need this variable if it is used only in one place?

Comment (kwiknick) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs]: I pulled it into a local variable because it's being used on line 113 and 115.

Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs]: Maybe I was not accurate. This variable is under the `#if` directive and actually used once in the result code.

**Comment by lzybkr** [comment]
Maybe we don't even need the year, that would be even simpler.

**Comment by kwiknick** [comment]
I can change it to not have a year, but I was trying to fulfill what was asked in Issue #3148.  There are no concerns of legal implications if the date isn't in there?


________________________________

**Comment by iSazonov** [comment]
Currently all MSFT applications use a year in copyright (cmd.exe, notepad.exe, skype...)

**Comment by lzybkr** [comment]
But they do not use a dynamic year - e.g. `cmd.exe` says `(c) 2016 Microsoft Corporation.` on my machine right now.

**Comment by iSazonov** [comment]
It seems that this (year) is tied to the main release time. I agree that is _internal_ MSFT pattern. Perhaps it has historical legal roots.

**Comment by andschwa** [comment]
Aye, we need @joeyaiello and @SteveL-MSFT to weigh in on the legal question.

**Comment by SteveL-MSFT** [comment]
I'll follow up on this.

**Comment by kwiknick** [comment]
Thanks for all the comments and feedback.  I out of habit brought my fork up to upstream/master, which, as it's supposed to, kicked off the CI builds.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
Waiting on legal

**Comment by joeyaiello** [comment]
Just so I understand, is the date dynamically generated at build time or runtime? If I had to guess, I think runtime is probably problematic, but that's just a hunch. 

Either way, I'm sending off the mail to legal now

**Comment by SteveL-MSFT** [comment]
@joeyaiello I've already started a conversation with legal

**Comment by kwiknick** [comment]
@joeyaiello It is generated at runtime, which I can see how/why It'd be problematic.  If legal decides that they want the date, I'll get it to generate at build time.  I should've done that in the first place.

**Comment by daxian-dbw** [comment]
Just found that `dotnet publish` with the RC4 dotnet CLI doesn't have the year in coryright banner now:
```
PS:53> dotnet publish -f netcoreapp1.0 -c release
Microsoft (R) Build Engine version 15.1.545.13942
Copyright (C) Microsoft Corporation. All rights reserved.

  console -> X:\arena\apps\console\bin\release\netcoreapp1.0\console.dll
```

**Comment by joeyaiello** [comment]
We'll ask if that's an option to legal as well

**Comment by iSazonov** [comment]
For legal too: What copyright we should use in the files? These files be published under MIT. There are also new files.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
We are approved to just use:

// Copyright (C) Microsoft Corporation. All rights reserved.

**Comment by kwiknick** [comment]
@SteveL-MSFT That's great news!  I'll update it this evening and push the changes.  Thanks everyone for all the collaboration.

**Comment by kwiknick** [comment]
@SteveL-MSFT the changes have been made and this is ready for review.  Thanks for being so helpful.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs [hunk 2]
```diff
@@ -109,9 +112,9 @@ public int Start(string consoleFilePath, string[] args)
             try
             {
 #if CORECLR
-                    var banner = ManagedEntranceStrings.ShellBannerNonWindowsPowerShell;
+                var banner = string.Format(ManagedEntranceStrings.ShellBannerNonWindowsPowerShell);
 #else
-                var banner = ManagedEntranceStrings.ShellBanner;
+                var banner = string.Format(ManagedEntranceStrings.ShellBanner);
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs]: string.Format - It seems should be removed.

And why we use Core and non-Core banner?

**Comment by kwiknick** [comment]
@SteveL-MSFT  Thanks for catching the unnecessary string.Formats.  I forgot to remove them as well a couple using statements.   The way I read the code when I first forked it, was that it shows a different banner based on the OS of the machine.  Do you want me to remove that as well and just have it to be the same message regardless?

**Comment by iSazonov** [comment]
I remember that @lzybkr open PR to remove Windows PowerShell from the repo so I believe we should remove the non-Core banner.

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@iSazonov I believe that PR is to just stop building Windows PowerShell.  I would leave the non-Core banner for now.

**Comment by iSazonov** [comment]
I agree.

**Comment by kwiknick** [comment]
Closed and Reopened to kick off the cla bot.

**Review by lzybkr — APPROVED** [review]

**Pull request merged**